metabase/metabase
# fix timeline events render on top of series

Closes https://github.com/metabase/metabase/issues/40180

### Description

We used ECharts marklines for both timeline events and the goal line. While the goal line should be rendered on top of series, timeline events must be behind. ECharts does not allow specifying custom `z` values for different marklines, it can be done only globally for all mark lines. This pull request reimplements the goal line with a custom series so that we can specify any z-index for it making in render on top of series.

### How to verify

- Create a bar chart with timeline events and the goal line
- Ensure timeline events are behind series while the goal line is above

### Demo

<img width="1370" alt="Screenshot 2024-04-02 at 4 50 08 PM" src="https://github.com/metabase/metabase/assets/14301985/b1a47f1c-3db4-4167-82a8-1e0ef012c6cc">

### Checklist
